frontend/leaderboard/src/Main.re

```diff
@@ -1,71 +1,48 @@
 /*
-  Main.re is the entry point of the leaderboard project.
+ Main.re is the entry point of the leaderboard project.
 
-  Main.re has the responsibilities for reading in a directory of blocks and
-  packing it up to be handed to Metrics.re. Blocks are defined in a json format.
-  The parsed fields for blocks are defined in Types/NewBlock.
+ Main.re has the responsibilities for querying the archive postgres database for
+ all the blockchain data and parsing the rows into blocks.
 
-  Additionally, Main.re expects to have the credentials and spreadsheet id
-  available in the form of environment variables. If no blocks are found,
-  the execution fails and reports an error.
- */
+ Additionally, Main.re expects to have the credentials, spreadsheet id, and postgres
+ connection string available in the form of environment variables.  */
 
-let blockDirectory =
-  ([%bs.node __dirname] |> Belt.Option.getExn |> Filename.dirname)
-  ++ "/src/blocks/";
-
-[@bs.val]
-external credentials: Js.Undefined.t(string) =
-  "process.env.GOOGLE_APPLICATION_CREDENTIALS";
-
-[@bs.val]
-external spreadsheetId: Js.Undefined.t(string) = "process.env.SPREADSHEET_ID";
-
-let blocks =
-  blockDirectory
-  |> Node.Fs.readdirSync
-  |> Array.map(file => {
-       let fileContents = Node.Fs.readFileAsUtf8Sync(blockDirectory ++ file);
-       let blockData = Js.Json.parseExn(fileContents);
-       let block = Types.NewBlock.unsafeJSONToNewBlock(blockData);
-       block.data.newBlock;
-     });
-
-let totalBlocks =
-  blockDirectory |> Node.Fs.readdirSync |> Array.length |> string_of_int;
-
-let setSheetsCredentials = () => {
-  switch (
-    Js.undefinedToOption(credentials),
-    Js.undefinedToOption(spreadsheetId),
-  ) {
-  | (Some(validCredentials), Some(spreadsheetId)) =>
-    Node.Fs.writeFileAsUtf8Sync(
-      "./google_sheets_credentials.json",
-      validCredentials,
-    );
-    Node.Process.putEnvVar(
-      "GOOGLE_APPLICATION_CREDENTIALS",
-      "./google_sheets_credentials.json",
-    );
-    Ok(spreadsheetId);
-  | (None, _) => Error("Invalid credentials environment variable")
-  | (_, None) => Error("Invalid spreadsheet environment variable")
+let getEnvOrFail = name =>
+  switch (Js.Dict.get(Node.Process.process##env, name)) {
+  | Some(value) => value
+  | None => failwith({j|Couldn't find env var: `$name`"|j})
   };
-};
 
-let main = () => {
-  switch (setSheetsCredentials()) {
-  | Ok(spreadsheetId) =>
-    blocks
-    |> Metrics.calculateMetrics
-    |> UploadLeaderboardPoints.uploadChallengePoints(spreadsheetId);
+let credentials = getEnvOrFail("GOOGLE_APPLICATION_CREDENTIALS");
+let spreadsheetId = getEnvOrFail("SPREADSHEET_ID");
+let pgConn = getEnvOrFail("PGCONN");
```

Comment:
can we change this to "pgConnection"? this is to connect to the postgres database right? i would just spell out the word to avoid confusion since Conn is not an automatically clear abbreviation

Reply:
or postgresConnection? something to make it clearer that it's referring to postgres